llvm/lib/CodeGen/MIRPrinter.cpp

```diff
@@ -728,7 +728,7 @@ void MIPrinter::print(const MachineBasicBlock &MBB) {
     HasLineAttributes = true;
   }
 
-  if (HasLineAttributes)
+  if (HasLineAttributes && !MBB.instrs().empty())
```

Comment:
Makes sense to me: you only need the blank line separator if there are some successors/liveins *and* some instructions.

Reply:
> Makes sense to me: you only need the blank line separator if there are some successors/liveins _and_ some instructions.

`HasLineAttributes` should already contain successors and liveins.

Reply:
> Can simplify to MBB.empty?

Yes! Done.